frontend/javascripts/oxalis/model/bucket_data_handling/pullqueue.ts

```diff
@@ -118,13 +122,41 @@ class PullQueue {
       if (!(error instanceof DOMException && error.name === "AbortError")) {
         // AbortErrors are deliberate. Don't show them on the console.
         console.error(error);
+        hasErrored = true;
       }
     } finally {
+      if (hasErrored) {
+        this.consecutiveErrorCount++;
+      } else {
+        this.consecutiveErrorCount = 0;
+      }
       this.batchCount--;
-      this.pull();
+
+      if (!hasErrored) {
+        // Continue to process the pull queue without delay.
+        this.pull();
+      } else {
+        // The current batch failed and we schedule a retry. However,
+        // parallel batches might fail, too, and also schedule a retry.
+        // To avoid that pull() is called X times in Y seconds, we only
+        // initiate a retry after a sleep if no concurrent invocation
+        // "claimed" the `isRetryScheduled` boolean.
+        if (!this.isRetryScheduled) {
+          this.isRetryScheduled = true;
+          sleep(this.getRetryDelay()).then(() => {
+            this.isRetryScheduled = false;
+            this.pull();
+          });
+        }
+      }
     }
   }
 
+  private getRetryDelay(): number {
+    const exponentialBackOff = 25 * 2 ** (this.consecutiveErrorCount / 10);
```

Comment:
this was just a gut feeling and turned okay during testing, I think. I'm open to suggestions, though.